janus-idp/operator
# Update existing resources when CR changes

## Description
Update the deployed resources when backstage CR is updated.

## Which issue(s) does this PR fix or relate to
- Fixed #72 

## PR acceptance criteria
- [X ] Tests

## How to test changes / Special notes to the reviewer
1) Deploy the operator on OpenShift and examples/bs1.yaml.
2) Update the CR spec to include:
```
spec:
   application:
      replicas: 2
```
and verify that the deployed backstage has 2 pods running.
3) Update the CR spec to include:
```
spec:
   application:
      extraEnvs:
        envs:
        - name: myenv1
          value: my-value-1
        - name: myenv2
          value: my-value-2
```
and check that the deployed backstage pod has env vars myenv1 and myenv2 set in the main container.
4) Update the CR spec to disable the route:
```
spec:
   application:
      route:
        enabled: true
```
and verify that the backstage route bas been cleaned up.
5)  Update the CR spec to disable the localDB:
```
spec:
  database:
    enableLocalDb: false
    authSecretName:  existing-secret
```
and verify that the local db statefulset, services and secret that were deployed earlier are now deleted.

Note that two status condition types "RouteSynced" and "LocalDbSynced" are introduced to support the functionality required.
